src/gluonts/shell/sagemaker/__init__.py

```diff
@@ -35,8 +35,8 @@ class DataConfig(BaseModel):
 
 
 class TrainEnv:
-    def __init__(self, path: Path = Path("/opt/ml")) -> None:
-        self.path = TrainPaths(path)
+    def __init__(self, path: TrainPaths = TrainPaths(Path("/opt/ml"))) -> None:
+        self.path = path
```

Comment:
the default value might be redundant because we already have a default in the click.option
```
@click.option(
    "--data-path",
    type=click.Path(exists=True),
    envvar="SAGEMAKER_DATA_PATH",
    default="/opt/ml",
    help="The root path of all folders mounted by the SageMaker runtime.",
)
```